### PR TITLE
Introduced basic_entity type

### PR DESCRIPTION
**Pull request by cjdb**

### CODE REVIEW

File: include/doge/entity/basic_entity.hpp [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+            break;
+         case cardinality::west:
+            strafe(left, t);
+            break;
+         }
+      }
+
+      void pitch(angle const a) noexcept
+      {
+         pitch_ = std::clamp(pitch() + a, -89.0_deg, 89.0_deg);
```
Comment (CallumHoward) [inline include/doge/entity/basic_entity.hpp]: magic numbers, make const member vars

File: include/doge/entity/basic_entity.hpp [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+      basic_entity(vec3 const& position, float const speed)
+         : position_{position},
+           speed_{speed}
+      {}
+
+      ~basic_entity() = default;
+
+      static inline constexpr ranges::Regular default_speed = 10.0f;
+   private:
+      vec3 position_{0.0f, 0.0f, 3.0f};
```
Comment (CallumHoward) [inline include/doge/entity/basic_entity.hpp:121]: Is `vec3` necessary? Can `vec2` be used instead?

Comment (CallumHoward) [reply include/doge/entity/basic_entity.hpp:121]: I also like to comment which header it comes from next to the includes, it looks like `glm/vec3.hpp` was removed

File: include/doge/entity/basic_entity.hpp [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+   enum class cardinality {
+      north,
+      south,
+      east,
+      west
+   };
+
+   class basic_entity {
+      enum dir_t { // should probably clean this up
```
Comment (CallumHoward) [inline include/doge/entity/basic_entity.hpp:41]: Gonna clean it up then?

File: include/doge/entity/basic_entity.hpp (hunk 2 above)
Comment (cjdb) [reply include/doge/entity/basic_entity.hpp:121]: Needs to be `vec3`, as it occupies 3D space. All GLM types have been annexed by `"doge/types.hpp"`: they're now `doge::vec3`, etc., so the implementation can change without affecting user code.

File: include/doge/entity/basic_entity.hpp (hunk 1 above)
Comment (cjdb) [reply include/doge/entity/basic_entity.hpp]: Fixed.

**Review by CallumHoward — APPROVED** [review]

**Pull request merged**